### PR TITLE
Add Property2, Property3, Property4 and some Int Properties

### LINKED ISSUE #6: add Property2, Property3 and Property4
to allow for testing properties for > 1 argument.

The current workaround would be to generate a tuple of things to test against and deconstruct them in the property code.

### PR DESCRIPTION
**Pull request by mfelsche**
Closes #6

### CODE REVIEW

File: ponycheck/test/forall.pony [hunk 1]
```diff
@@ -17,3 +25,45 @@ class MultipleForAllTest is UnitTest
...
+  fun name(): String => "ponycheck/for_all2"
+
+  fun apply(h: TestHelper) ? =>
+    Ponycheck.for_all2[U8, String](
+      recover Generators.unit[U8](0) end,
+      recover Generators.ascii() end,
+      h)(
+        {(arg1, arg2, h) =>
+          h.assert_false(arg2.contains(String.from_array([as U8: arg1])))
+          })?
```
Comment (Theodus) [inline ponycheck/test/forall.pony]: Minor nit: the closing brace for lambda should be at the same indentation level as the opening one

File: ponycheck/test/forall.pony [hunk 2]
```diff
@@ -17,3 +25,45 @@ class MultipleForAllTest is UnitTest
...
+    Ponycheck.for_all4[U8, U8, U8, String](
+      recover Generators.unit[U8](0) end,
+      recover Generators.u8() end,
+      recover Generators.u8() end,
+      recover Generators.ascii() end,
+      h)(
+        {(b1, b2, b3, str, h) =>
+          let cmp = String.from_array([b1; b2; b3])
+          h.assert_false(str.contains(cmp))
+          })?
```
Comment (Theodus) [inline ponycheck/test/forall.pony]: see the comment above

File: ponycheck/test/ints.pony [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      property_notify,
+      property_logger,
+      h.env)
+    runner.run()
+
+class SuccessfulIntPairProperty is IntPairProperty
+  fun name(): String => "property/intpair/property"
+
+  fun int_property[T: (Int & Integer[T] val)](x: T, y: T, h: PropertyHelper) =>
+    h.assert_eq[T](x * y, y* x)
```
Comment (Theodus) [inline ponycheck/test/ints.pony]: Minor nit: infix operator spacing, should be `y * x`

**Review by Theodus — APPROVED** [review]
Looks good to me. Just a few minor formatting comments

**Comment by mfelsche** [comment]
Thank you very much @Theodus !

**Pull request merged**